htmx-spring-boot/src/main/java/io/github/wimdeblauwe/htmx/spring/boot/mvc/HtmxReswap.java

```diff
@@ -0,0 +1,279 @@
+ * Represents a HX-Reswap response header value.
+ *
+ * @see <a href="https://htmx.org/attributes/hx-swap/">hx-swap</a>
+ */
+public class HtmxReswap {
+
+    private final HxSwapType type;
+    private Duration swap;
+    private Duration settle;
+    private Position scroll;
+    private String scrollTarget;
+    private Position show;
+    private String showTarget;
+    private boolean transition;
+    private Boolean focusScroll;
+
+    /**
+     * Insert the response before the first child of the target element.
+     */
+    public static HtmxReswap afterBegin() {
+        return new HtmxReswap(HxSwapType.AFTER_BEGIN);
+    }
+
+    /**
+     * Insert the response after the target element.
+     */
+    public static HtmxReswap afterEnd() {
+        return new HtmxReswap(HxSwapType.AFTER_END);
+    }
+
+    /**
+     * Insert the response before the target element.
+     */
+    public static HtmxReswap beforeBegin() {
+        return new HtmxReswap(HxSwapType.BEFORE_BEGIN);
+    }
+
+    /**
+     * Insert the response after the last child of the target element.
+     */
+    public static HtmxReswap beforeEnd() {
+        return new HtmxReswap(HxSwapType.BEFORE_END);
+    }
+
+    /**
+     * Deletes the target element regardless of the response.
+     */
+    public static HtmxReswap delete() {
+        return new HtmxReswap(HxSwapType.DELETE);
+    }
+
+    /**
+     * Replace the inner html of the target element.
+     */
+    public static HtmxReswap innerHtml() {
+        return new HtmxReswap(HxSwapType.INNER_HTML);
+    }
+
+    /**
+     * Does not append the response to target element (out of band items will still be processed).
+     */
+    public static HtmxReswap none() {
+        return new HtmxReswap(HxSwapType.NONE);
+    }
+
+    /**
+     * Replace the entire target element with the response.
+     */
+    public static HtmxReswap outerHtml() {
+        return new HtmxReswap(HxSwapType.OUTER_HTML);
+    }
+
+    HtmxReswap(HxSwapType type) {
+        this.type = type;
+    }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+        if (object == null || getClass() != object.getClass()) {
+            return false;
+        }
+        HtmxReswap that = (HtmxReswap) object;
+        return transition == that.transition && Objects.equals(type, that.type) && Objects.equals(swap, that.swap) && Objects.equals(settle, that.settle) && scroll == that.scroll && Objects.equals(scrollTarget, that.scrollTarget) && show == that.show && Objects.equals(
+                showTarget, that.showTarget) && Objects.equals(focusScroll, that.focusScroll);
+    }
+
+    /**
+     * Enable or disable auto-scrolling to focused inputs between requests.
+     *
+     * @param enabled {@code true} if auto-scrolling to focused inputs between requests is enabled, otherwise {@code false}
+     * @return self
+     */
+    public HtmxReswap focusScroll(boolean enabled) {
```

Comment:
The order of the methods in the class could be a bit more consistent:

- I propose to put equals, hashcode and toString method together
- Move this `focusScroll` builder method with the other builder methods.

Reply:
I automatically sort the methods by visibility and in alphabetical order. But if you prefer, I can sort them manually.